server/datastore/mysql/software.go

```diff
@@ -215,9 +215,10 @@ var dialect = goqu.Dialect("mysql")
 func listSoftwareDB(
 	ctx context.Context, q sqlx.QueryerContext, hostID *uint, opts fleet.SoftwareListOptions,
 ) ([]fleet.Software, error) {
-	ds := dialect.From(goqu.I("host_software").As("hs")).SelectDistinct(
+	ds := dialect.From(goqu.I("host_software").As("hs")).Select(
 		"s.*",
 		goqu.COALESCE(goqu.I("scp.cpe"), "").As("generated_cpe"),
+		goqu.COUNT(goqu.DISTINCT("hs.host_id")).As("host_count"),
```

Comment:
If `listSoftwareDB` is called with `hostID != nil` (when listing software of a particular host only), then `host_count` will always be `1`. Is that expected?

Reply:
While I can think of a use for knowing how common a software is from the host detail look, that's outside of the scope of this work. It bleeds into the other APIs because of the struct.

If we count more than just the host, should we count across the whole list of hosts, or just the team for the host? I think it needs a bit more thinking if we ever need it.

Reply:
+1, it's just something worth mentioning on the API docs.

Reply:
Done!